databricks_cli/pipelines/cli.py

```diff
@@ -59,8 +69,20 @@ def deploy_cli(api_client, spec_arg, spec):
         raise RuntimeError('The spec should be provided either by an option or argument')
     src = spec_arg if bool(spec_arg) else spec
     spec_obj = _read_spec(src)
+    if 'id' not in spec_obj:
+        pipeline_id = str(uuid.uuid4())
+        click.echo("Updating spec with id: {}".format(pipeline_id))
+        spec_obj['id'] = pipeline_id
+        _write_spec(src, spec_obj)
+    elif not _validate_pipeline_id(spec_obj['id']):
+        error_and_quit("Pipeline id has invalid characters")
```

Comment:
Say what's allowed and also print the id they provided. Also, maybe the error_and_quit should just be inside the _validate_pipeline_id call so you don't have to repeat those bits?